tests/storages_tests/rdb_tests/test_storage.py

```diff
@@ -312,3 +320,43 @@ def test_upgrade_distributions(optuna_version: str) -> None:
             new_study.optimize(objective_test_upgrade_distributions, n_trials=1)
 
         storage.engine.dispose()  # Be sure to disconnect db
+
+
+def test_create_new_trial_with_retries() -> None:
+    storage = RDBStorage("sqlite:///:memory:")
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+
+    n_retries = 0
+
+    def mock_func(
```

Comment:
I checked `_get_prepared_new_trial` and confirmed that the routine is identical.